### PR TITLE
feat(timeout): add optional timeout when waiting for completion

### PR DESCRIPTION
**Pull request by Adir-Shemesh**

### CODE REVIEW

File: intezer_sdk/analysis.py [hunk 1]
```diff
@@ -66,26 +69,33 @@ def send(self, wait: typing.Union[bool, int] = False) -> None:
...
             if sleep_before_first_check:
                 time.sleep(interval)
             status_code = self.check_status()
 
-            while status_code != consts.AnalysisStatusCode.FINISH:
+            timeout_passed = timeout and datetime.datetime.now() - start_time > timeout
+            while not timeout_passed and status_code != consts.AnalysisStatusCode.FINISH:
                 time.sleep(interval)
                 status_code = self.check_status()
+                timeout_passed = timeout and datetime.datetime.now() - start_time > timeout
```
Comment (kfirstri) [inline intezer_sdk/analysis.py:98]: i wonder if we should raise something or alert the user somehow? it feels that if someone set a timeout eh expects to "catch" it somehow,

Comment (kfirstri) [reply intezer_sdk/analysis.py:98]: relevant also to all the other places that wait_timeout has been added, we should talk about the behavior

Comment (Adir-Shemesh) [reply intezer_sdk/analysis.py]: raised TimeoutError on timeouts

File: intezer_sdk/analysis.py [hunk 2]
```diff
@@ -66,16 +69,21 @@ def send(self, wait: typing.Union[bool, int] = False) -> None:
...
+                            interval: int = None,
+                            sleep_before_first_check=False,
+                            timeout: typing.Optional[datetime.timedelta] = None):
         """
         Blocks until the analysis is completed
         :param interval: The interval to wait between checks
         :param sleep_before_first_check: Whether to sleep before the first status check 
+        :param timeout: Maximum duration to wait for analysis completion
         """
+        start_time = datetime.datetime.now()
```
Comment (davidt99) [inline intezer_sdk/analysis.py:86]: as a best practice - always use `datetime.utcnow()` and not `datetime.now()`

**Review by davidt99 — CHANGES_REQUESTED** [review]
Add change to the changelog and docs

File: intezer_sdk/analysis.py (hunk 2 above)
Comment (Adir-Shemesh) [reply intezer_sdk/analysis.py]: done and done

**Review by davidt99 — APPROVED** [review]

File: CHANGES [hunk 3]
```diff
@@ -3,6 +3,10 @@ Not released
 - Add family search
```
Comment (kfirstri) [inline CHANGES:3]: ok this changes is old.. the family search is already released, so you can remove it..
another thing is can you add also the update to README.md?

**Review by kfirstri — APPROVED** [review]

**Pull request merged**